mitmproxy/tools/console/consoleaddons.py

```diff
@@ -383,6 +383,7 @@ def edit_focus(self, part: str) -> None:
             part in ("response-headers", "response-body", "set-cookies") and
             flow.response is None
         )
+        flow.backup()
```

Comment:
Nit: Don't move it between `require_dummy_response` definition and usage, this can live above or below :)